# 6.1 Data Cleaning & Wrangling

## This script contain following points:

### 1. Importing the datasets

### 2. Data Cleaning

### 2. Data Wrangling

### 3. Basic descriptive analysis

## 01. Import libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

## 02. Import datasets

In [2]:
# Create path
path = r'C:\Users\Quinn\Documents\CF - Data Analysis\Data Immersion\Achieve 6\AirBnB in Europe citites Project'

In [3]:
# Import all the orginal datasets
# Amsterdam datasets
ams_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'amsterdam_weekdays.csv'))
ams_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'amsterdam_weekends.csv'))

# Athens datasets
athens_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'athens_weekdays.csv'))
athens_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'athens_weekends.csv'))

# Barcelona datasets
barc_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'barcelona_weekdays.csv'))
barc_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'barcelona_weekends.csv'))

# Berlin datasets
berlin_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'berlin_weekdays.csv'))
berlin_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'berlin_weekends.csv'))

# Budapest datasets
buda_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'budapest_weekdays.csv'))
buda_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'budapest_weekends.csv'))

# Lisbon datasets
lis_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'lisbon_weekdays.csv'))
lis_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'lisbon_weekends.csv'))

# London datasets
london_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'london_weekdays.csv'))
london_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'london_weekends.csv'))

# Paris datasets
paris_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'paris_weekdays.csv'))
paris_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'paris_weekends.csv'))

# Rome datasets
rome_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'rome_weekdays.csv'))
rome_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'rome_weekends.csv'))

# Vienna datasets
vienna_weekdays = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'vienna_weekdays.csv'))
vienna_weekends = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'vienna_weekends.csv'))

In [7]:
# Adding new columns 'weekday/weekday' and 'city' to each dataframe
# weekday/weekday: marked as 'weekday' if data is recorded on weekdays, 'weekend' if data is recored on weekend
# city: object, capitalized city names
paris_weekdays['weekday/weekend'], paris_weekdays['city'] = 'Weekday', 'Paris'
paris_weekends['weekday/weekend'], paris_weekends['city'] = 'Weekend', 'Paris'

rome_weekdays['weekday/weekend'], rome_weekdays['city'] = 'Weekday', 'Rome'
rome_weekends['weekday/weekend'], rome_weekends['city'] = 'Weekend', 'Rome'

vienna_weekdays['weekday/weekend'], vienna_weekdays['city'] = 'Weekday', 'Vienna'
vienna_weekends['weekday/weekend'], vienna_weekends['city'] = 'Weekend', 'Vienna'

ams_weekdays['weekday/weekend'], ams_weekdays['city'] = 'Weekday', 'Amsterdam'
ams_weekends['weekday/weekend'], ams_weekends['city'] = 'Weekend', 'Amsterdam'

athens_weekdays['weekday/weekend'], athens_weekdays['city'] = 'Weekday', 'Athens'
athens_weekends['weekday/weekend'], athens_weekends['city'] = 'Weekend', 'Athens'

barc_weekdays['weekday/weekend'], barc_weekdays['city'] = 'Weekday', 'Barcelona'
barc_weekends['weekday/weekend'], barc_weekends['city'] = 'Weekend', 'Barcelona'

berlin_weekdays['weekday/weekend'], berlin_weekdays['city'] = 'Weekday', 'Berlin'
berlin_weekends['weekday/weekend'], berlin_weekends['city'] = 'Weekend', 'Berlin'

buda_weekdays['weekday/weekend'], buda_weekdays['city'] = 'Weekday', 'Budapest'
buda_weekends['weekday/weekend'], buda_weekends['city'] = 'Weekend', 'Budapest'

lis_weekdays['weekday/weekend'], lis_weekdays['city'] = 'Weekday', 'Lisbon'
lis_weekends['weekday/weekend'], lis_weekends['city'] = 'Weekend', 'Lisbon'

london_weekdays['weekday/weekend'], london_weekdays['city'] = 'Weekday', 'London'
london_weekends['weekday/weekend'], london_weekends['city'] = 'Weekend', 'London'

In [8]:
# random check one weekday dataframe and one weekend dataframe
lis_weekdays

Unnamed: 0     realSum        room_type  room_shared  room_private  \
0              0  138.133208     Private room        False          True   
1              1  124.061914     Private room        False          True   
2              2  194.183865     Private room        False          True   
3              3  191.604128  Entire home/apt        False         False   
4              4  327.861163  Entire home/apt        False         False   
...          ...         ...              ...          ...           ...   
2852        2852  263.133208  Entire home/apt        False         False   
2853        2853  272.279550  Entire home/apt        False         False   
2854        2854  529.784240  Entire home/apt        False         False   
2855        2855  118.198874     Private room        False          True   
2856        2856  234.052533  Entire home/apt        False         False   

      person_capacity  host_is_superhost  multi  biz  cleanliness_rating  ...  \
0                 2.0               True      1    0                10.0  ...   
1                 2.0               True      1    0                10.0  ...   
2                 4.0               True      1    0                10.0  ...   
3                 4.0              False      0    1                 9.0  ...   
4                 6.0              False      1    0                 9.0  ...   
...               ...                ...    ...  ...                 ...  ...   
2852              4.0              False      0    1                10.0  ...   
2853              4.0              False      0    0                10.0  ...   
2854              6.0              False      0    1                10.0  ...   
2855              2.0               True      1    0                10.0  ...   
2856              2.0              False      1    0                10.0  ...   

          dist  metro_dist  attr_index  attr_index_norm   rest_index  \
0     4.328029    0.298484   74.230170         2.450656   176.868292   
1     4.465504    0.293603   72.570845         2.395875   173.909352   
2     4.475232    0.167860   72.517895         2.394127   177.996137   
3     0.850978    0.589445  409.695507        13.525805   806.463205   
4     0.667029    0.530351  537.015471        17.729183   775.718380   
...        ...         ...         ...              ...          ...   
2852  1.644997    0.757184  157.885745         5.212485   349.315504   
2853  1.868345    0.866777  142.044110         4.689485   310.053294   
2854  0.330145    0.265451  425.231552        14.038716  1342.732529   
2855  2.295936    1.236675  125.115410         4.130596   255.799483   
2856  2.315085    1.755327  113.991588         3.763351   297.839518   

      rest_index_norm      lng       lat  weekday/weekend    city  
0            7.910210 -9.14034  38.75137          Weekday  Lisbon  
1            7.777875 -9.14092  38.75260          Weekday  Lisbon  
2            7.960651 -9.14245  38.75264          Weekday  Lisbon  
3           36.068044 -9.13000  38.71100          Weekday  Lisbon  
4           34.693021 -9.13200  38.71300          Weekday  Lisbon  
...               ...      ...       ...              ...     ...  
2852        15.622693 -9.12374  38.72042          Weekday  Lisbon  
2853        13.866740 -9.12141  38.72130          Weekday  Lisbon  
2854        60.052009 -9.14300  38.71100          Weekday  Lisbon  
2855        11.440307 -9.11770  38.72390          Weekday  Lisbon  
2856        13.320495 -9.16491  38.70582          Weekday  Lisbon  

[2857 rows x 22 columns]

In [9]:
barc_weekends

Unnamed: 0     realSum        room_type  room_shared  room_private  \
0              0  532.036214  Entire home/apt        False         False   
1              1  169.897829     Private room        False          True   
2              2  196.895292     Private room        False          True   
3              3  141.271208     Private room        False          True   
4              4  715.200037  Entire home/apt        False         False   
...          ...         ...              ...          ...           ...   
1273        1273  327.460609     Private room        False          True   
1274        1274  242.977169     Private room        False          True   
1275        1275  138.943841     Private room        False          True   
1276        1276  185.258454     Private room        False          True   
1277        1277  254.614006     Private room        False          True   

      person_capacity  host_is_superhost  multi  biz  cleanliness_rating  ...  \
0                 4.0              False      0    1                10.0  ...   
1                 2.0               True      1    0                10.0  ...   
2                 2.0              False      1    0                 9.0  ...   
3                 3.0              False      0    1                 9.0  ...   
4                 4.0               True      0    1                 9.0  ...   
...               ...                ...    ...  ...                 ...  ...   
1273              2.0               True      0    0                 9.0  ...   
1274              2.0              False      0    0                 9.0  ...   
1275              2.0              False      0    0                10.0  ...   
1276              2.0              False      0    0                 2.0  ...   
1277              2.0              False      0    0                10.0  ...   

          dist  metro_dist  attr_index  attr_index_norm   rest_index  \
0     1.111998    0.630504  526.471472        20.327672   915.591455   
1     1.751831    0.124013  320.128264        12.360522   794.279660   
2     0.621731    0.297516  503.354025        19.435081  1242.434221   
3     1.648311    0.089442  342.595820        13.228021   820.921958   
4     0.693560    0.357653  537.803135        20.765201  1081.560116   
...        ...         ...         ...              ...          ...   
1273  2.608873    0.932342  271.642736        10.488440   537.512628   
1274  1.859134    0.555944  343.045595        13.245387   679.561854   
1275  2.670450    0.968673  266.091811        10.274112   534.080063   
1276  2.266090    0.387429  290.526738        11.217573   627.521382   
1277  2.394845    0.954950  284.900718        11.000346   564.678656   

      rest_index_norm      lng       lat  weekday/weekend       city  
0           20.112468  2.17556  41.39624          Weekend  Barcelona  
1           17.447656  2.14906  41.38714          Weekend  Barcelona  
2           27.292106  2.16500  41.38300          Weekend  Barcelona  
3           18.032897  2.15342  41.37912          Weekend  Barcelona  
4           23.758242  2.17498  41.39214          Weekend  Barcelona  
...               ...      ...       ...              ...        ...  
1273        11.807347  2.19672  41.39929          Weekend  Barcelona  
1274        14.927691  2.19099  41.39269          Weekend  Barcelona  
1275        11.731945  2.19738  41.39954          Weekend  Barcelona  
1276        13.784536  2.19679  41.39029          Weekend  Barcelona  
1277        12.404093  2.19590  41.39637          Weekend  Barcelona  

[1278 rows x 22 columns]

In [10]:
# Combine all data of each city
paris_all = paris_weekdays.merge(paris_weekends, how = 'outer')
rome_all = rome_weekdays.merge(rome_weekends, how = 'outer')
vienna_all = vienna_weekdays.merge(vienna_weekends, how = 'outer')
ams_all = ams_weekdays.merge(ams_weekends, how = 'outer')
athens_all = athens_weekdays.merge(athens_weekends, how = 'outer')
barc_all = barc_weekdays.merge(barc_weekends, how = 'outer')
berlin_all = berlin_weekdays.merge(berlin_weekends, how = 'outer')
buda_all = buda_weekdays.merge(buda_weekends, how = 'outer')
lis_all = lis_weekdays.merge(lis_weekends, how = 'outer')
london_all = london_weekdays.merge(london_weekends, how = 'outer')

In [11]:
# Combine all cities into one dataset
airbnb = paris_all.merge(rome_all, how = 'outer'
                ).merge(vienna_all, how = 'outer'
                ).merge(ams_all, how = 'outer'
                ).merge(athens_all, how = 'outer'
                ).merge(barc_all, how = 'outer'
                ).merge(berlin_all, how = 'outer'
                ).merge(buda_all, how = 'outer'
                ).merge(lis_all, how = 'outer'
                ).merge(london_all, how = 'outer')

In [12]:
# Checking the new dataframe
airbnb

Unnamed: 0      realSum        room_type  room_shared  room_private  \
0               0   296.159940     Private room        False          True   
1               1   288.237487     Private room        False          True   
2               2   211.343089     Private room        False          True   
3               3   298.956100  Entire home/apt        False         False   
4               4   247.926181  Entire home/apt        False         False   
...           ...          ...              ...          ...           ...   
51702        5374  1035.066676     Private room        False          True   
51703        5375   181.095510     Private room        False          True   
51704        5376   420.517886  Entire home/apt        False         False   
51705        5377   217.314612     Private room        False          True   
51706        5378   489.193067  Entire home/apt        False         False   

       person_capacity  host_is_superhost  multi  biz  cleanliness_rating  \
0                  2.0               True      0    0                10.0   
1                  2.0               True      0    0                10.0   
2                  2.0              False      0    0                10.0   
3                  2.0              False      0    1                 9.0   
4                  4.0              False      0    0                 7.0   
...                ...                ...    ...  ...                 ...   
51702              2.0              False      0    1                10.0   
51703              2.0              False      1    0                10.0   
51704              2.0              False      0    0                10.0   
51705              2.0              False      0    0                10.0   
51706              2.0              False      0    0                 9.0   

       ...      dist  metro_dist  attr_index  attr_index_norm   rest_index  \
0      ...  0.699821    0.193709  518.478947        25.239380  1218.662228   
1      ...  2.100005    0.107221  873.216962        42.507907  1000.543327   
2      ...  3.302325    0.234724  444.556077        21.640840   902.854467   
3      ...  0.547567    0.195997  542.142014        26.391291  1199.184166   
4      ...  1.197921    0.103573  406.928958        19.809165  1070.775497   
...    ...       ...         ...         ...              ...          ...   
51702  ...  5.155373    0.883829  271.191201        18.851558   528.042180   
51703  ...  5.072177    0.344055  268.165870        18.641255   554.906232   
51704  ...  6.709982    0.926188  181.892001        12.644022   395.397885   
51705  ...  6.736940    0.353634  184.278328        12.809905   471.029678   
51706  ...  4.174953    0.610443  434.954210        30.235363   697.673227   

       rest_index_norm      lng       lat  weekday/weekend    city  
0            71.608028  2.35385  48.86282          Weekday   Paris  
1            58.791463  2.32436  48.85902          Weekday   Paris  
2            53.051310  2.31714  48.87475          Weekday   Paris  
3            70.463506  2.35600  48.86100          Weekday   Paris  
4            62.918272  2.35915  48.86648          Weekday   Paris  
...                ...      ...       ...              ...     ...  
51702         9.451035 -0.05357  51.51255          Weekend  London  
51703         9.931855 -0.05620  51.51823          Weekend  London  
51704         7.076933 -0.22172  51.52180          Weekend  London  
51705         8.430610 -0.22286  51.49715          Weekend  London  
51706        12.487135 -0.06739  51.50889          Weekend  London  

[51707 rows x 22 columns]

## 03. Data Cleaning

### Mixed type data

In [13]:
# Checking mixed typed
for col in airbnb.columns.tolist():
  weird = (airbnb[[col]].map(type) != airbnb[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (airbnb[weird]) > 0:
    print (col)

#### There is no mixed type data column in the final dataframe

### Missing values

In [14]:
# Checking missing values
airbnb.isnull().sum()

Unnamed: 0                    0
realSum                       0
room_type                     0
room_shared                   0
room_private                  0
person_capacity               0
host_is_superhost             0
multi                         0
biz                           0
cleanliness_rating            0
guest_satisfaction_overall    0
bedrooms                      0
dist                          0
metro_dist                    0
attr_index                    0
attr_index_norm               0
rest_index                    0
rest_index_norm               0
lng                           0
lat                           0
weekday/weekend               0
city                          0
dtype: int64

#### There is no missing value found in the dataframe

### Duplicates

In [15]:
# Checking duplicates 
airbnb_dups = airbnb[airbnb.duplicated()]

In [16]:
airbnb_dups

Empty DataFrame
Columns: [Unnamed: 0, realSum, room_type, room_shared, room_private, person_capacity, host_is_superhost, multi, biz, cleanliness_rating, guest_satisfaction_overall, bedrooms, dist, metro_dist, attr_index, attr_index_norm, rest_index, rest_index_norm, lng, lat, weekday/weekend, city]
Index: []

[0 rows x 22 columns]

#### There is no duplicates found the in the dataframe

## 04. Data Wrangling

In [19]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51707 entries, 0 to 51706
Data columns (total 22 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Unnamed: 0                  51707 non-null  int64  
 1   realSum                     51707 non-null  float64
 2   room_type                   51707 non-null  object 
 3   room_shared                 51707 non-null  bool   
 4   room_private                51707 non-null  bool   
 5   person_capacity             51707 non-null  float64
 6   host_is_superhost           51707 non-null  bool   
 7   multi                       51707 non-null  int64  
 8   biz                         51707 non-null  int64  
 9   cleanliness_rating          51707 non-null  float64
 10  guest_satisfaction_overall  51707 non-null  float64
 11  bedrooms                    51707 non-null  int64  
 12  dist                        51707 non-null  float64
 13  metro_dist                  517

In [21]:
# Dropping unnecessary columns
airbnb_drop = airbnb.drop(columns = ['Unnamed: 0','attr_index','rest_index','lng','b'])

#### I drop columns 'Unamed: 0', 'lng', 'lat' becasue it's not relevent with my analysis. For columns 'attr_index','rest_index', I already used accordingly normalized columns so I don't need these columns anymore

In [22]:
airbnb_drop

realSum        room_type  room_shared  room_private  \
0       296.159940     Private room        False          True   
1       288.237487     Private room        False          True   
2       211.343089     Private room        False          True   
3       298.956100  Entire home/apt        False         False   
4       247.926181  Entire home/apt        False         False   
...            ...              ...          ...           ...   
51702  1035.066676     Private room        False          True   
51703   181.095510     Private room        False          True   
51704   420.517886  Entire home/apt        False         False   
51705   217.314612     Private room        False          True   
51706   489.193067  Entire home/apt        False         False   

       person_capacity  host_is_superhost  multi  biz  cleanliness_rating  \
0                  2.0               True      0    0                10.0   
1                  2.0               True      0    0                10.0   
2                  2.0              False      0    0                10.0   
3                  2.0              False      0    1                 9.0   
4                  4.0              False      0    0                 7.0   
...                ...                ...    ...  ...                 ...   
51702              2.0              False      0    1                10.0   
51703              2.0              False      1    0                10.0   
51704              2.0              False      0    0                10.0   
51705              2.0              False      0    0                10.0   
51706              2.0              False      0    0                 9.0   

       guest_satisfaction_overall  bedrooms      dist  metro_dist  \
0                            97.0         1  0.699821    0.193709   
1                            97.0         1  2.100005    0.107221   
2                            94.0         1  3.302325    0.234724   
3                            91.0         1  0.547567    0.195997   
4                            82.0         1  1.197921    0.103573   
...                           ...       ...       ...         ...   
51702                       100.0         0  5.155373    0.883829   
51703                       100.0         1  5.072177    0.344055   
51704                        98.0         1  6.709982    0.926188   
51705                        94.0         1  6.736940    0.353634   
51706                        92.0         1  4.174953    0.610443   

       attr_index_norm  rest_index_norm weekday/weekend    city  
0            25.239380        71.608028         Weekday   Paris  
1            42.507907        58.791463         Weekday   Paris  
2            21.640840        53.051310         Weekday   Paris  
3            26.391291        70.463506         Weekday   Paris  
4            19.809165        62.918272         Weekday   Paris  
...                ...              ...             ...     ...  
51702        18.851558         9.451035         Weekend  London  
51703        18.641255         9.931855         Weekend  London  
51704        12.644022         7.076933         Weekend  London  
51705        12.809905         8.430610         Weekend  London  
51706        30.235363        12.487135         Weekend  London  

[51707 rows x 17 columns]

In [23]:
# Rename columns for better understanding
airbnb_drop.rename(columns = {'realSum' : 'price'}, inplace = True)
airbnb_drop.rename(columns = {'multi' : 'multi_listing'}, inplace = True)
airbnb_drop.rename(columns = {'biz' : 'business_listing'}, inplace = True)
airbnb_drop.rename(columns = {'dist' : 'city_center_dist'}, inplace = True)

In [25]:
# Final dataframe info
airbnb_drop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51707 entries, 0 to 51706
Data columns (total 17 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   price                       51707 non-null  float64
 1   room_type                   51707 non-null  object 
 2   room_shared                 51707 non-null  bool   
 3   room_private                51707 non-null  bool   
 4   person_capacity             51707 non-null  float64
 5   host_is_superhost           51707 non-null  bool   
 6   multi_listing               51707 non-null  int64  
 7   business_listing            51707 non-null  int64  
 8   cleanliness_rating          51707 non-null  float64
 9   guest_satisfaction_overall  51707 non-null  float64
 10  bedrooms                    51707 non-null  int64  
 11  city_center_dist            51707 non-null  float64
 12  metro_dist                  51707 non-null  float64
 13  attr_index_norm             517

## 05. Basic descriptive analysis

In [27]:
airbnb_drop.describe()

price  person_capacity  multi_listing  business_listing  \
count  51707.000000     51707.000000   51707.000000      51707.000000   
mean     279.879591         3.161661       0.291353          0.350204   
std      327.948386         1.298545       0.454390          0.477038   
min       34.779339         2.000000       0.000000          0.000000   
25%      148.752174         2.000000       0.000000          0.000000   
50%      211.343089         3.000000       0.000000          0.000000   
75%      319.694287         4.000000       1.000000          1.000000   
max    18545.450285         6.000000       1.000000          1.000000   

       cleanliness_rating  guest_satisfaction_overall     bedrooms  \
count        51707.000000                51707.000000  51707.00000   
mean             9.390624                   92.628232      1.15876   
std              0.954868                    8.945531      0.62741   
min              2.000000                   20.000000      0.00000   
25%              9.000000                   90.000000      1.00000   
50%             10.000000                   95.000000      1.00000   
75%             10.000000                   99.000000      1.00000   
max             10.000000                  100.000000     10.00000   

       city_center_dist    metro_dist  attr_index_norm  rest_index_norm  
count      51707.000000  51707.000000     51707.000000     51707.000000  
mean           3.191285      0.681540        13.423792        22.786177  
std            2.393803      0.858023         9.807985        17.804096  
min            0.015045      0.002301         0.926301         0.592757  
25%            1.453142      0.248480         6.380926         8.751480  
50%            2.613538      0.413269        11.468305        17.542238  
75%            4.263077      0.737840        17.415082        32.964603  
max           25.284557     14.273577       100.000000       100.000000

#### Prices range from around €34.78 to €18,545.45. The median price (50th percentile) is approximately €211.34
#### Person capacity ranges from 2 to 6. The mean person capacity is approximately 3.16, indicating that the typical listing accommodates around three people.
#### About 29.1% of the listings are multi-listings (have multiple listings from the same host).
#### Approximately 35.0% of the listings are designated as business listings.
#### Cleanliness ratings range from 2 to 10. The average cleanliness rating is 9.39 out of 10, and the data is right-skewed.
#### Overall ratings range from 20 to 100. The average overall guest satisfaction rating is approximately 92.63 out of 100, and the data is right-skewed.
#### The number of bedrooms ranges from 0 to 10. The average number of bedrooms is approximately 1.16. The data is lightly right-skewed
#### The average distance from the city centre is approximately 3.19 kilometres.
#### The average distance from the nearest metro station is approximately 0.68 kilometres.

## 06. Export as pickel file

In [28]:
airbnb_drop.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'airbnb_clean.pkl'))